libs/dart/vault_storages/lib/src/profiles/vfs_profile_repository.dart

```diff
@@ -74,6 +75,8 @@ class VfsProfileRepository implements ProfileRepository {
     _keyStorage = configuration.keyStorage!;
 
     _rootKeyPair = await _getRootKeyPair();
+    _accountVaultDataManagerService =
+        await _createAccountVaultDataManagerService();
```

Comment:
```suggestion
    _accountVaultDataManagerService = await _memoizedDataManagerService(
      walletKeyId: _rootAccountKeyId,
      kek: Uint8List(2),
    );
```

Would this work?
it would remove the need for `_createAccountVaultDataManagerService` definition

As for the implementation there is a slight performance change, 
it now creates the service during configuration whereas previous implementation was only creating the service on first usage.